src/diffusers/models/modeling_utils.py

```diff
@@ -671,10 +671,6 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
             hf_quantizer = None
 
         if hf_quantizer is not None:
-            if device_map is not None:
-                raise NotImplementedError(
-                    "Currently, `device_map` is automatically inferred for quantized models. Support for providing `device_map` as an input will be added in the future."
-                )
```

Comment:
@sayakpaul I'm not sure how this impacts BnB quantizer. I assume it was disabled for BnB for some reason I'm not aware of. It works with TorchAO as expected though so if you need this to have some kind of guard for torchao-specific, I'll add it

Reply:
cc @SunMarc

Reply:
> How did you test that?

There is a test for this in `tests/quantization/torchao/test_torchao.py` called `test_offload` that can be used to verify that cpu/disk offloading works with torchao

Reply:
Yeah but this is about custom user-provided `device_map`s. What am I missing?

Reply:
yeah that works for me.